scalameta/trees/shared/src/main/scala/scala/meta/internal/trees/ParentChecks.scala

```diff
@@ -105,6 +105,15 @@ object ParentChecks {
     tree.tpe.is[Type.Singleton] ==> (parent.is[Ctor.Secondary] && destination == "init")
   }
 
+  def EnumCase(tree: Tree, parent: Tree, destination: String): Boolean = {
+    parent.is[Template]
+  }
+
+  def EnumTemplate(tree: Template, parent: Tree, destination: String): Boolean = {
+    val hasEnumCase = tree.stats.exists(st => st.is[Defn.EnumCase] || st.is[Defn.RepeatedEnumCase])
```

Comment:
Added the two checks here, which together make sure that EnumCase is in Template and if template contains enum case then the parent can only be Enum. @olafurpg waht do you think?

Reply:
We will need to go through all stats, so this might not be perfect :thinking: